docs/configuration/config-file-reference.md

```diff
@@ -871,6 +871,10 @@ The `querier_config` configures the Cortex querier.
 # CLI flag: -querier.ingester-streaming
 [ingester_streaming: <boolean> | default = true]
 
+# Use streaming RPCs for metadata APIs from ingester.
+# CLI flag: -querier.ingester-metadata-streaming
+[ingester_metadata_streaming: <boolean> | default = true]
```

Comment:
I am pondering on if we should have a new flag or just use the `ingester_streaming` flag. What's your thought on this?

Reply:
> I am pondering on if we should have a new flag or just use the ingester_streaming flag. What's your thought on this?

We need a way to migrate... so i think a new flag is needed indeed...

But i agree if we have more APIs that will be converted to streaming, we should do all at once.

Reply:
I've added streaming in LabelNames and LabelValues as well in this PR. This should reduce the number of new flags we need for live migration.

Reply:
If the default value of `ingester_metadata_streaming` is true, won't a lot of customers gets auto migrate to metadata stream?

Reply:
Good catch. I can change that. :)